Month3/Project1/Project1/TDinamicArray.h

```diff
@@ -0,0 +1,290 @@
+#pragma once
+#include "Square.h"
+#include <stdlib.h>     
+#include <time.h> 
+
+
+template <typename T>
+class TDinamicArray
+{
+private:
+	int Capacity = 3;
+	int Size = 0;
+	T* Array = new T[Capacity];
+public:
+	TDinamicArray() 
+	{
+	
+	}
+	TDinamicArray(const int EntryCapacity)
+	{
+		Array = new T[EntryCapacity];
+	
+	}
+	~TDinamicArray() 
+	{
+		delete[] Array;
+	}
+
+	TDinamicArray(const TDinamicArray<T>& Array1)
+	{
+		Append(Array1);
+	}
+	
+
+	T& operator[](int Index)
+	{
+		return Array[Index];
+	}
+	const T& operator[](int Index)const
+	{
+		return Array[Index];
+	}
+
+
+
+	T& Front()
+	{
+		return Array[0];
+	}
+	const T& Front()const 
+	{
+		return Array[0];
+	}
+
+
+	T& Back()
+	{	
+		return Array[Size];
+	}
+	const T& Back() const
+	{
+		return Array[Size];
+	}
+
+
+	T& GetData()
+	{
+		return Array;
+	}
+	const T& GetData() const
+	{
+		return Array;
+	}
+
+	bool Empty()
```

Comment:
I think this should be named `IsEmpty()` instead of `Empty()`. The former I would think checks to see if it is empty, while the latter sounds like I would want to empty (or clear) the array.